spanner/ddl/ast_test.go

```diff
@@ -79,24 +79,35 @@ func TestPrintCreateTable(t *testing.T) {
 	cds["col1"] = ColumnDef{Name: "col1", T: Type{Name: Int64}, NotNull: true}
 	cds["col2"] = ColumnDef{Name: "col2", T: Type{Name: String, Len: MaxLength}, NotNull: false}
 	cds["col3"] = ColumnDef{Name: "col3", T: Type{Name: Bytes, Len: int64(42)}, NotNull: false}
-	ct := CreateTable{
+	t1 := CreateTable{
 		"mytable",
 		[]string{"col1", "col2", "col3"},
 		cds,
 		[]IndexKey{IndexKey{Col: "col1", Desc: true}},
 		nil,
 		"",
+		"",
+	}
+	t2 := CreateTable{
+		"mytable",
+		[]string{"col1", "col2", "col3"},
+		cds,
+		[]IndexKey{IndexKey{Col: "col1", Desc: true}},
+		nil,
+		"parent",
+		"",
 	}
 	tests := []struct {
 		name       string
 		protectIds bool
 		expected   string
+		ct         CreateTable
 	}{
-		{"no quote", false, "CREATE TABLE mytable (col1 INT64 NOT NULL, col2 STRING(MAX), col3 BYTES(42)) PRIMARY KEY (col1 DESC)"},
-		{"quote", true, "CREATE TABLE `mytable` (`col1` INT64 NOT NULL, `col2` STRING(MAX), `col3` BYTES(42)) PRIMARY KEY (`col1` DESC)"},
+		{"no quote", false, "CREATE TABLE mytable (col1 INT64 NOT NULL, col2 STRING(MAX), col3 BYTES(42)) PRIMARY KEY (col1 DESC)", t1},
+		{"quote", true, "CREATE TABLE `mytable` (`col1` INT64 NOT NULL, `col2` STRING(MAX), `col3` BYTES(42)) PRIMARY KEY (`col1` DESC),\nINTERLEAVE IN PARENT `parent` ON DELETE CASCADE", t2},
```

Comment:
In general, we try to limit each test case to just testing one thing. That makes it easier to pinpoint issues when they come up, such as changes that break tests.

In this case, I'd suggest adding a third test case, and leaving the other two unchanged i.e.

{"no quote", false, "CREATE TABLE mytable (col1 INT64 NOT NULL, col2 STRING(MAX), col3 BYTES(42)) PRIMARY KEY (col1 DESC)", t1},
{"quote", true, "CREATE TABLE `mytable` (`col1` INT64 NOT NULL, `col2` STRING(MAX), `col3` BYTES(42)) PRIMARY KEY (`col1` DESC)", t1},
{"interleaved", false, "CREATE TABLE mytable (col1 INT64 NOT NULL, col2 STRING(MAX), col3 BYTES(42)) PRIMARY KEY (col1 DESC),\nINTERLEAVE IN PARENT parent ON DELETE CASCADE", t2},

Reply:
updated tests, separated interleaved test